src/map/Map.js

```diff
@@ -338,8 +340,21 @@ L.Map = L.Evented.extend({
 
 		padding = L.point(padding || [0, 0]);
 
+		if (snap <= 0) {

```

Comment:
I know adding code is not nice, but when the zoom level is not snapped, bounding boxes should fit exactly to the map bounds (or at least I'd expect them to).